### ISSUE TITLE
Clarification on the use of `jnp.log` in loss calculation

### ISSUE THREAD

**Issue by cyrus-baker** [dkweiss31/optamiqs#12]
In the following code:
```python
def loss(
    parameters: Array | dict,
    costs: Cost,
    model: Model,
    method: Method,
    gradient: Gradient,
    dq_options: dq.Options,
) -> [float, Array]:
    result, H = model(parameters, method, gradient, dq_options)
    cost_values, terminate = zip(*costs(result, H, parameters), strict=True)
    total_cost = jax.tree.reduce(jnp.add, cost_values)
    total_cost = jnp.log(jnp.sum(jnp.asarray(total_cost)))
    expects = result.expects if hasattr(result, 'expects') else None
    return total_cost, (total_cost, cost_values, terminate, expects)
```

I found `jnp.log` is applied to the `jnp.sum(jnp.asarray(total_cost))`, I am curious about the necessity of this log function. 
In my simulation, loss function without `jnp.log` cooperate better with `optax.adam` with the default parameters (b1, b2, ...). 

I guess because the Adam optimizer has the second-order momentum. However, due to the `jnp.log` function, the grads is scaled with `1/L`. when loss `L` is very small, the grads scale very fast, the adam optimizer can not catch it. 

I wonder is there any special design with this `jnp.log`? If not, could we remove it?

**Comment by dkweiss31** [comment]
The motivation here is when trying to optimize gates for very low infidelities. The idea is you want the optimizer to be just as excited to get you from 1e-3 to 1e-4 infidelity as it was to get you from 1e-2 to 1e-3. My understanding (you can correct me if I am incorrect!) is that without the log, getting from 1e-3 to 1e-4 is 10x less exciting to the optimizer as it was for 1e-2 to 1e-3. I am happy to reconsider this choice if you can show me a small MWE where making this change leads to substantially better results.

**Comment by cyrus-baker** [comment]
I have tested sevel different tasks and find they have different behavior. But basically, they have similar performance when reaching 1e-4 when using **second order optimizer** (Adam, lbfgs, ...). Different learning rate, different initial guess will both change "which one converge faster". As `adam` has become the most wildly used optimizer, this may not be necessary.
